### PR TITLE
TAS-2155/fixed-TS-errors-part3

### PR DESCRIPTION
**Pull request by marjan-ara**

### CODE REVIEW

**Comment by satanmourner** [comment]
@marjan-ara please resolve conflicts

File: src/modules/Jobs/components/applyModalQuestions/index.tsx [hunk 1]
```diff
@@ -56,8 +56,8 @@ const ApplyModalQuestions: React.FC<ApplyModalQuestionsProps> = ({ answers, setA
               })}
               onChange={type => handleSelectOption(question.id, Number(type.value), type.label)}
               errors={
-                questionErrors.find(e => e.id === question.id)
-                  ? [questionErrors.find(e => e.id === question.id).message]
+                questionErrors.find(e => e.id === question.id)?.message
```
Comment (satanmourner) [inline src/modules/Jobs/components/applyModalQuestions/index.tsx:59]: no need to first line
`[questionErrors.find(e => e.id === question.id)?.message || '']` is enough:
1- if message found -> message
2- if message not found -> undefined || '' -> would be empty string

Comment (satanmourner) [reply src/modules/Jobs/components/applyModalQuestions/index.tsx:59]: line 71, 72 too

File: src/modules/Search/components/ResultList/ResultList.types.ts [hunk 2]
```diff
@@ -1,7 +1,5 @@
-interface SearchItem {
-  id: string;
-  title: string;
-}
+import { SearchItem } from '../../containers/SearchModal/SearchModal.types';
```
Comment (satanmourner) [inline src/modules/Search/components/ResultList/ResultList.types.ts:1]: please use an absolute path starting with `src/...`

File: src/modules/Search/components/ResultList/index.tsx [hunk 3]
```diff
@@ -5,14 +5,14 @@ import { Chip } from 'src/modules/general/components/Chip';
 
 import css from './result-list.module.scss';
 import { ResultListProps } from './ResultList.types';
-import { Item } from '../../containers/SearchModal/SearchModal.types';
+import { SearchItem } from '../../containers/SearchModal/SearchModal.types';
```
Comment (satanmourner) [inline src/modules/Search/components/ResultList/index.tsx:8]: absolute path

File: src/modules/general/components/SearchDropdown/index.tsx [hunk 4]
```diff
@@ -85,7 +86,7 @@ export const SearchDropdown: React.FC<SelectProps> = ({
   ...props
 }) => {
   const selectedVal = props.value || '';
-  const selectRef = useRef<HTMLSelectElement>(null);
+  const selectRef = useRef<any>(null);
```
Comment (satanmourner) [inline src/modules/general/components/SearchDropdown/index.tsx:89]: instead use `Select<any>` and Select is from 'react-select`

File: src/modules/general/components/input/input.tsx [hunk 5]
```diff
@@ -78,11 +78,11 @@ export const Input: React.FC<InputProps> = ({
           {postfix}
         </InputAdornment>
       )}
-      {postfixDropdown && (
+      {postfixDropdown && postfixDropdown.onChange && (
```
Comment (satanmourner) [inline src/modules/general/components/input/input.tsx:81]: FMI? when we don't have `onChange`, we won't show `InputDropdown`?

File: src/modules/feeds/feedItem/index.tsx [hunk 6]
```diff
@@ -106,7 +106,7 @@ const FeedItem: React.FC<FeedItemProps> = ({
   };
   const editPostData = {
     postId,
-    cause: Object.values(SOCIAL_CAUSES).find(social => social.label === cause) || null,
+    cause: Object.values(SOCIAL_CAUSES).find(social => social.label === cause),
```
Comment (satanmourner) [inline src/modules/feeds/feedItem/index.tsx:109]: FMI: no default value?

**Review by satanmourner — CHANGES_REQUESTED** [review]

File: src/pages/impact/useImpact.tsx [hunk 7]
```diff
@@ -11,20 +11,17 @@ export const useImpact = () => {
...
-        .filter(item => item.offer !== null)
+        .filter(item => !!item.offer)
         .map(item => {
-          if (item.offer) {
-            if ((item?.offer?.currency && ['USD', 'YEN'].includes(item?.offer?.currency)) || item.offer.currency) {
-              return { type: 'paid', hours: item.offer.total_hours };
-            } else {
-              return { type: 'volunteered', hours: item.offer.total_hours };
-            }
+          if ((item?.offer?.currency && ['USD', 'YEN'].includes(item?.offer?.currency)) || item.offer.currency) {
```
Comment (satanmourner) [inline src/pages/impact/useImpact.tsx:21]: no need to ensure about `offer` because you filter defined and non nullable `item.offer` before

**Review by satanmourner — CHANGES_REQUESTED** [review]

File: src/modules/Jobs/components/applyModalQuestions/index.tsx (hunk 1 above)
Comment (marjan-ara) [reply src/modules/Jobs/components/applyModalQuestions/index.tsx:59]: The `errors` is an array or undefined. In this case, if message not found, [''] is returned, while we want an undefined

File: src/modules/general/components/SearchDropdown/index.tsx (hunk 4 above)
Comment (marjan-ara) [reply src/modules/general/components/SearchDropdown/index.tsx:89]: It doesn't resolves ts error. Can you try it?

**Review by satanmourner — APPROVED** [review]

**Pull request merged**